runtime/src/bank.rs

```diff
@@ -1845,29 +1845,25 @@ impl Bank {
         &self,
         signature: &Signature,
     ) -> Option<Result<()>> {
-        if let Some(status) = self.get_signature_confirmation_status(signature) {
-            if status.slot == self.slot() {
-                return Some(status.status);
+        if let Some((slot, status)) = self.get_signature_status_slot(signature) {
+            if slot <= self.slot() {
+                return Some(status);
             }
         }
         None
     }
 
-    pub fn get_signature_confirmation_status(
-        &self,
-        signature: &Signature,
-    ) -> Option<SignatureConfirmationStatus<Result<()>>> {
+    pub fn get_signature_status_slot(&self, signature: &Signature) -> Option<(Slot, Result<()>)> {
         let rcache = self.src.status_cache.read().unwrap();
-        rcache.get_signature_status_slow(signature, &self.ancestors)
+        rcache.get_signature_slot(signature, &self.ancestors)
```

Comment:
Just fyi, this `self.ancestors` will only include up to the root `R` at the time this bank was created right: https://github.com/solana-labs/solana/blob/master/runtime/src/bank.rs#L478-L481.

So if the signature is requested from a rooted bank from before `R`, then the status cache will only the last `MAX_RECENT_BLOCKHASHES` of roots. Hopefully this is ok for downstream consumers?

Reply:
Yes, that is definitely expected. The historical signature-status search (#8823 , blocked by the Blockstore rework we've been talking about) is intended for anything older than the status cache.
Sure, I can update the documentation for now.